### PR TITLE
testing user namespace update

### PR DESCRIPTION
**Pull request by xcoulon**
E2E tests for https://github.com/codeready-toolchain/host-operator/pull/230

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: test/e2e/nstemplatetier_test.go [hunk 1]
```diff
@@ -92,47 +92,106 @@ func TestNSTemplateTiers(t *testing.T) {
...
-	defer ctx.Cleanup()
-
-	// Create and approve "testing-template-updates" user
-	murName := "testingtemplateupdates"
-	userSignup := createAndApproveSignup(t, awaitility, murName)
+func TestUpdateNSTemplateTier(t *testing.T) {
+	// this test verifies that `maxPoolSize` TemplateUpdateRequests are created when the `cheesecake` NSTemplateTier is updated
+	// with new templates and there are MUR accounts associated with this tier.
+	ctx, awaitility := testsupport.WaitForDeployments(t, &toolchainv1alpha1.NSTemplateTier{})
+	// defer ctx.Cleanup()
```
[the remark this answers is not in the document]
Comment (xcoulon) [reply test/e2e/nstemplatetier_test.go:99]: yes, I had commented it out while testing, but I'll restore it

File: test/e2e/nstemplatetier_test.go [hunk 2]
```diff
@@ -92,47 +92,106 @@ func TestNSTemplateTiers(t *testing.T) {
...
+			UntilMasterUserRecordHasCondition(provisioned())) // ignore other conditions, such as notification sent, etc.
+		require.NoError(t, err)
+		syncIndexes[i] = mur.Spec.UserAccounts[0].SyncIndex
+		t.Logf("'%s' initial syncIndex: '%s'", mur.Name, syncIndexes[i])
+	}
+	// finally, let's retrieve the `advanced` NSTemplateTier
+	advancedTier := &toolchainv1alpha1.NSTemplateTier{}
+	err = hostAwaitility.Client.Get(context.TODO(), types.NamespacedName{
+		Namespace: hostAwaitility.Ns,
+		Name:      "advanced",
```
[the remark this answers is not in the document]
Comment (xcoulon) [reply test/e2e/nstemplatetier_test.go:159]: ok, I will add this kind of test as well

[the remark this answers is not in the document]
Comment (xcoulon) [reply test/e2e/nstemplatetier_test.go]: so, in 5513753 I've refactored the code (using funcs to setup, update and verify the different resources), and also, I've used 2 groups of users, using 2 distinct tiers, one updated with the "advanced" templates, and the one with the "team" templates. The good thing in this refactoring is that both tiers are updated and only after the resources are verified, so we also make sure that there's no "mixing" ;)

File: test/e2e/nstemplatetier_test.go [hunk 3]
```diff
@@ -92,47 +92,148 @@ func TestNSTemplateTiers(t *testing.T) {
...
+	updateTemplateTier(t, awaitility, cheesecakeTier, "advanced")
+	// when updating the "cookie" tier with the "team" template refs (ie, different number of namespaces)
+	updateTemplateTier(t, awaitility, cookieTier, "team")
 
-	// wait for the user to be provisioned for the first time
-	verifyResourcesProvisionedForSignup(t, awaitility, userSignup, "basic")
+	// then
+	verifyResourceUpdates(t, awaitility, cheesecakeSyncIndexes, cheesecakeTier.Name, "advanced")
+	verifyResourceUpdates(t, awaitility, cookieSyncIndexes, cookieTier.Name, "team")
+}
```
Comment (MatousJobanek) [inline test/e2e/nstemplatetier_test.go:118]: Do you think that we could also test updating the cookie tier back to basic to verify that the additional namespace will be created?
```suggestion
	// and when updating the "cookie" tier back to the "basic" template refs (ie, different number of namespaces)
	updateTemplateTier(t, awaitility, cookieTier, "basic")
	
	// then
	verifyResourceUpdates(t, awaitility, cookieSyncIndexes, cookieTier.Name, "basic")
}
```

**Comment by xcoulon** [comment]
> I really like this new format of the test - it's short and very easy to read and modify. Awesome improvement!!! 👍 🥇

Thanks!! 

> 
> I have one suggestion to add a case when the update would create a new namespace - something that is not covered yet.

good point! Let me do that ;)

**Pull request merged**